diff -aruN libdnf-0.63.0/docs/hawkey/conf.py libdnf-0.63.0-new/docs/hawkey/conf.py
--- libdnf-0.63.0/docs/hawkey/conf.py	2021-05-18 17:07:23.000000000 -0700
+++ libdnf-0.63.0-new/docs/hawkey/conf.py	2022-03-29 11:03:39.000000000 -0700
@@ -260,6 +260,6 @@
 rst_prolog = """
 .. default-domain:: py
 .. _libsolv: https://github.com/openSUSE/libsolv
-.. _bugzilla: https://bugzilla.redhat.com/enter_bug.cgi?product=Fedora&component=hawkey
+.. _bugzilla: https://bugs.openela.org/
 
 """
diff -aruN libdnf-0.63.0/libdnf/conf/Const.hpp libdnf-0.63.0-new/libdnf/conf/Const.hpp
--- libdnf-0.63.0/libdnf/conf/Const.hpp	2021-05-18 17:07:23.000000000 -0700
+++ libdnf-0.63.0-new/libdnf/conf/Const.hpp	2022-03-29 11:03:47.000000000 -0700
@@ -41,7 +41,7 @@
                  "installonlypkg(vm)",
                  "multiversion(kernel)"};
 
-constexpr const char * BUGTRACKER="https://bugzilla.redhat.com/enter_bug.cgi?product=Fedora&component=dnf";
+constexpr const char * BUGTRACKER="https://bugs.openela.org/";
 
 }
 
